li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaFederatedProducerImpl.java

```diff
@@ -355,7 +185,35 @@ public void close() {
 
   @Override
   public void close(long timeout, TimeUnit timeUnit) {
-    throw new UnsupportedOperationException("Not implemented yet");
+    if (_producers.isEmpty()) {
+      LOG.warn("No producers to close for cluster group {}", _clusterGroup);
+      return;
+    }
+
+    LOG.info("Closing LiKafkaProducer for cluster group {} in {} {}...", _clusterGroup, timeout, timeUnit);
+
+    long startTimeMs = System.currentTimeMillis();
+    long deadlineTimeMs = startTimeMs + timeUnit.toMillis(timeout);
+    CountDownLatch countDownLatch = new CountDownLatch(_producers.entrySet().size());
+    for (Map.Entry<ClusterDescriptor, LiKafkaProducer<K, V>> entry : _producers.entrySet()) {
+      Thread taskThread = new Thread(() -> {
```

Comment:
nit: you could get away without a local variable

Reply:
Done